src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -356,7 +357,7 @@ public TracerSettings(IConfigurationSource source)
         /// Default is <c>Datadog</c>
         /// <seealso cref="ConfigurationKeys.Propagators"/>
         /// </summary>
-        public HashSet<PropagatorType> Propagators { get; set; }
+        public HashSet<string> Propagators { get; set; }
```

Comment:
I feel that `PropagatorType` can be preserved here and plugins could have their own enums. We just need to give plugins a way to parse the tracer configuration on their own.

Reply:
Hmm can you elaborate on this? I'm not sure I understand how we could mix and match enums

Reply:
I think that `TracerSettings` should only contain "core" settings.
The "core" should not know anything about the plugins apart from the contract (extension point interface).
The plugins can parse the env vars to create their own settings.
For example, if the user would set `OTEL_PROPAGATORS=tracecontext,baggage,splunk` then the "core" should return two enums for both w3c propagators `tracecontext,baggage` and do NOP for the third one `splunk`. The third `splunk` one can be then processed by the plugin. The plugin should simply return an additional propagator when this value is set.
MOre: https://github.com/open-telemetry/opentelemetry-specification/blob/b46bcab5fb709381f1fd52096a19541370c7d1b3/specification/sdk-environment-variables.md#general-sdk-configuration

Reply:
I could not find any good way to preserve enum type. I think the same logic should use the same design line, so some of the base logic could be reused.

Reply:
I think it makes more sense to create an issue for it and tackle it later (but not too late) if nobody finds an easy way how to address the issue now.

Reply:
I think it is good as it is 👍  Resolving

Reply:
Yeah, didn't like it either but let's look at it separately.